os/linux.go

```diff
@@ -231,17 +232,35 @@ func (c Linux) FileExist(h Host, path string) bool {
 	return h.Execf(`test -e %s 2> /dev/null`, escape.Quote(path), exec.Sudo(h)) == nil
 }
 
-// LineIntoFile tries to find a matching line in a file and replace it with a new entry
-// TODO refactor this into go because it's too magical.
+// LineIntoFile tries to find a line starting with the matcher and replace it with a new entry. If match isn't found, the string is appended to the file.
+// TODO add exec.Opts (requires modifying readfile and writefile signatures)
 func (c Linux) LineIntoFile(h Host, path, matcher, newLine string) error {
```

Comment:
Now that this method is no longer a "remote" method, I wonder if it should still remain part of rig in the first place. The use case is also quite narrowed down to k0sctl's needs, I'd say.

Reply:
Yeah maybe. I was also thinking that rig should have readfile + writefile that would use io.Reader / io.Writer. In that case it could do read => modify => write in two streams without reading the whole thing in memory at once. Launchpad also uses lineintofile but has implemented it locally.

There's still probably some simpler way to do the shell version. I don't like the copypaste here but didn't find a pleasing way to share such stuff between the os modules.